core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/ImmutablesTaster.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper.reflect.internal;
+
+import java.lang.annotation.Annotation;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.immutables.value.Value;
+import org.jdbi.v3.core.config.JdbiConfig;
+import org.jdbi.v3.core.generic.GenericTypes;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoProperties.PojoProperty;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+
+public class ImmutablesTaster implements Function<Type, Optional<PojoProperties<?>>>, JdbiConfig<ImmutablesTaster> {
+
+    static {
+        Value.Immutable.class.toString(); // Fail fast
+    }
+
+    private Set<Class<?>> registered = new HashSet<>();
+
+    public ImmutablesTaster() {}
+
+    private ImmutablesTaster(ImmutablesTaster other) {
+        registered.addAll(other.registered);
+    }
+
+    @Override
+    public ImmutablesTaster createCopy() {
+        return new ImmutablesTaster(this);
+    }
+
+    public void register(Class<?> iface, Class<?> impl) {
+        registered.add(iface);
+        registered.add(impl);
+    }
+
+    @Override
+    public Optional<PojoProperties<?>> apply(Type t) {
+        final Optional<Class<?>> defn =
+            Arrays.stream(GenericTypes.getErasedType(t).getInterfaces())
+            .filter(
+                i -> i.getAnnotation(Value.Immutable.class) != null
```

Comment:
Note that these checks are totally ineffective right now, none of the annotations have runtime retention...